sycamore/transforms/merge_elements.py

```diff
@@ -0,0 +1,130 @@
+        if element2[0].type == 'Title':
+            return False
+        if element1[1] + 1 + element2[1] > self.max_tokens:
+            return False
+        return True
+
+    def _merge(self, element1: Tuple[Element, int], element2: Tuple[Element, int]) -> Tuple[Element, int]:
+        """
+        Merge two elements; the new element's fields will be set as:
+        
+            type: "Section"
+            binary_representation: elt1.binary_representation + elt2.binary_representation
+            text_representation: elt1.text_representation + '\n' + elt2.text_representation
+            bbox: the minimal bbox that contains both elt1's and elt2's bboxes
+            properties: elt1's properties + any of elt2's properties that are not in elt1
+                > note: if elt1 and elt2 have different values for the same property, we take elt1's value
+            
+            > note: if any input field is None we take the other element's field without merge logic
+
+        Args:
+            element1 (Tuple[Element, int]): the first element (and number of tokens in it)
+            element2 (Tuple[Element, int]): the second element (and number of tokens in it)
+
+        Returns:
+            Tuple[Element, int]: a new merged element from the inputs (and number of tokens in it)
+        """
+        elt1, tok1 = element1
+        elt2, tok2 = element2
+        new_toks = 0
+        new_elt = Element()
+        new_elt.type = "Section"
+        # Merge binary representations by concatenation
+        if elt1.binary_representation == None or elt2.binary_representation == None:
+            new_elt.binary_representation = elt1.binary_representation or elt2.binary_representation
+        else:
+            new_elt.binary_representation = elt1.binary_representation + elt2.binary_representation
+        # Merge text representations by concatenation with a newline
+        if elt1.text_representation == None or elt2.text_representation == None:
+            new_elt.text_representation = elt1.text_representation or elt2.text_representation
+            new_toks = max(tok1, tok2)
+        else:
+            new_elt.text_representation = elt1.text_representation + '\n' + elt2.text_representation
+            new_toks = tok1 + 1 + tok2
+        # Merge bbox by taking the coords that make the largest box
+        if elt1.bbox == None and elt2.bbox == None:
+            pass
+        elif elt1.bbox == None or elt2.bbox == None:
+            new_elt.bbox = elt1.bbox or elt2.bbox
+        else:
+            new_elt.bbox = BoundingBox(
+                min(elt1.bbox.x1, elt2.bbox.x1),
+                min(elt1.bbox.y1, elt2.bbox.y1),
+                max(elt1.bbox.x2, elt2.bbox.x2),
+                max(elt1.bbox.y2, elt2.bbox.y2)
+            )
+        # Merge properties by taking the union of the keys
+        for k, v in elt1.properties.items():
+            new_elt.properties[k] = v
+        for k, v in elt2.properties.items():
```

Comment:
so element 2's properties are going to override? I think this merging is non-trivial, do we have an open issue for this

Reply:
element 2's properties get overwritten by element1's, yes. #105 is the issue for this feature; I didn't really know what to do with the keys. If you have a suggestion I'll take it